ci/Dockerfile

```diff
@@ -36,7 +36,7 @@ RUN dnf --enablerepo=powertools install -y scl-utils gcc-toolset-${TOOLSET_VERSI
 RUN mkdir -m 777 /usr/local/rapids /rapids
 
 # 3.22.3: CUDA architecture 'native' support + flexible CMAKE_<LANG>_*_LAUNCHER for ccache
-ARG CMAKE_VERSION=3.26.4
+ARG CMAKE_VERSION=3.28.6
```

Comment:
I'm not sure the update to cudf is actually required since libcudfjni does build with cmake 3.26.4.  It's only spark-rapids-jni that had issues with the older cmake due to the use of nvbench.

Reply:
At the very least for consistency's sake. There are plenty of moving parts already.

Reply:
https://github.com/rapidsai/cudf/pull/17342